lockable/allocation.py

```diff
@@ -52,3 +54,21 @@ def allocation_durations(self) -> timedelta:
         """
         end_time = self.release_time or datetime.now()
         return end_time - self.allocation_start_time
+
+    @staticmethod
+    def get_matching_resources(resource_list: [{}], requirements: {}) -> [{}]:
+        """ Get matching resources from resource list """
+        # if there is {<field>: {$exists: true}} in requirements, filter out resources
+        # that does not have the field. if value is false, filter out resources that have the field.
+        def in_filter(resource):
+            """ Check if resource matches requirements """
+            for key, value in requirements.items():
+                if isinstance(value, dict) and '$exists' in value:
+                    if value['$exists'] and key not in resource:
```

Comment:
How does this behave with input such as `{<field>: {$exists: 1}}`? Should it assert if value is not bool?

Reply:
yep, should ensure it's boolean value